applicable_pose_generator/src/pose_reachability_filter.cpp

```diff
@@ -0,0 +1,72 @@
+#include <urdf_model/model.h>
+
+#include <kdl/chainiksolverpos_lma.hpp>
+
+#include "ros/ros.h"
+
+#include "applicable_pose_generator/pose_reachability_filter.h"
+#include <boost/shared_ptr.hpp>
+
+#include <kdl_parser/kdl_parser.hpp>
+#include <eigen_conversions/eigen_kdl.h>
+
+
+namespace CreateChain
+{
+chain_creation::chain_creation()
+{
+  ros::NodeHandle pnh("~");
+  if (!pnh.getParam("robot_urdf", robot_urdf)){
+    ROS_ERROR("did not set parameter robot_urdf");
```

Comment:
whenever a parameter needs to be set, one can provide a default, or error out. This seems to go on without some necessary information